jetstream/kv.go

```diff
@@ -1271,6 +1278,55 @@ func (kv *kvs) ListKeys(ctx context.Context, opts ...WatchOpt) (KeyLister, error
 	return kl, nil
 }
 
+// KeysWithFilters retrieves keys that match the provided filters using WatchFiltered.
+func (kv *kvs) KeysWithFilters(ctx context.Context, _ KeyValue, filters ...string) ([]string, error) {
+	if len(filters) == 0 {
+		return nil, errors.New("filters cannot be empty")
+	}
+
+	watcher, err := kv.WatchFiltered(ctx, filters, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer watcher.Stop()
+
+	var keys []string
+	for entry := range watcher.Updates() {
+		if entry == nil { // Indicates all initial values are received
+			break
+		}
+		keys = append(keys, entry.Key())
+	}
+	return keys, nil
+}
+
+// ListKeysWithFilters returns a channel of keys matching the provided filters using WatchFiltered.
+func (kv *kvs) ListKeysWithFilters(ctx context.Context, _ KeyValue, filters ...string) (<-chan string, error) {
+	if len(filters) == 0 {
```

Comment:
This check is not needed, it should be possible to pass no filters to just get all keys. it will essentially be a way more versatile version of current `ListKeys()`.